runtime/src/main/scala/grpc/akkastreams/GrpcSinkStage.scala

```diff
@@ -0,0 +1,41 @@
+package grpc.akkastreams
+
+import java.util.concurrent.atomic.AtomicReference
+
+import akka.stream.{Attributes, Inlet, SinkShape}
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler}
+import io.grpc.stub.CallStreamObserver
+
+class GrpcSinkStage[I](observer: CallStreamObserver[I]) extends GraphStage[SinkShape[I]] {
+  val in = Inlet[I]("grpc.in")
+  override val shape: SinkShape[I] = SinkShape.of(in)
+  override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) with InHandler with Runnable {
```

Comment:
why the runnable, who invokes that run()? Could we instead materialize some value if we need to set things on this from the outside?

Reply:
The `Runnable` is invoked by the grpc framework when the grpc `StreamObserver` becomes ready to accept more messages.

I don't think it runs on the same thread as the stream logic ... hence the atomic variables.

It is set by ```reqStream.setOnReadyHandler(this)```. Not sure it should be exposed to the user though.